halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/GenerateCommand.java

```diff
@@ -1,4 +1,37 @@
+/*
+ * Copyright 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 package com.netflix.spinnaker.halyard.cli.command.v1;
 
-public class GenerateCommand {
+import com.netflix.spinnaker.halyard.cli.services.v1.Daemon;
+import com.netflix.spinnaker.halyard.cli.services.v1.DaemonService;
+import lombok.AccessLevel;
+import lombok.Getter;
+
+public class GenerateCommand extends NestableCommand {
+  @Getter(AccessLevel.PUBLIC)
+  private String commandName = "generate";
+
+  @Getter(AccessLevel.PUBLIC)
+  private String description = "Generate the full Spinnaker config for you current deployment.";
+
+  @Override
+  protected void executeThis() {
+    DaemonService service = Daemon.getService();
+    String currentDeployment = service.getCurrentDeployment();
+    service.generateDeployment(currentDeployment, "", false);
```

Comment:
I dont understand how this works. I dont think you discuss it in your design doc yet (the section "deploying and configuring" is TBD, but I didnt look over the rest to see if it is there).

I assume this is generating a deployment configuration and not actually deploying. In section 1.2 of your design doc you say the configuration is never stored on the machine running halyard, only staged there during "distribution of the configuration". I assume that is what we are doing here.

But it seems weird that you ask for the deployment and then generate it.  Why is current-deployment in the halconfig at all? Actually I have a few more questions about halconfig like why is the version information and configuration tied together? I understand that the configuration you write needs to tie together, but this is where the trying would happen. Otherwise it seems that changing a version requires a complete re-specification of the configuration rather than being able to reference a baseline and add some deltas to it (before eventually re-baselining it).